compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1486,25 +1486,24 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     let kind = key.descr();
                     let mut returned_async_output_error = false;
                     for sp in values {
-                        err.span_label(
-                            *sp,
-                            format!(
-                                "{}{}{} {}{}",
-                                if sp.is_desugaring(DesugaringKind::Async)
-                                    && !returned_async_output_error
-                                {
-                                    "checked the `Output` of this `async fn`, "
-                                } else if count == 1 {
-                                    "the "
-                                } else {
-                                    ""
-                                },
-                                if count > 1 { "one of the " } else { "" },
-                                target,
-                                kind,
-                                pluralize!(count),
-                            ),
-                        );
+                        if sp.is_desugaring(DesugaringKind::Async) && !returned_async_output_error {
+                            err.span_label(
+                                *sp,
+                                format!("{}", "calling an async function returns a future"),
```

Comment:
```suggestion
                                format!("{}", "async functions returns futures"),
```
Maybe this reads slightly more clearly?

Reply:
Went ahead and incorporated this, with slight different wording.

Reply:
And thank you for the suggestion!